src/components/Explore/ShareButton.tsx

```diff
@@ -0,0 +1,139 @@
+import { useOnClickOutside } from 'hooks/useOnClickOutside'
+import { darken } from 'polished'
+import { useRef, useState } from 'react'
+import { Check, Link, Share, Twitter } from 'react-feather'
+import { useModalIsOpen, useToggleModal } from 'state/application/hooks'
+import { ApplicationModal } from 'state/application/reducer'
+import styled, { useTheme } from 'styled-components/macro'
+
+const TWITTER_WIDTH = 560
+const TWITTER_HEIGHT = 480
+const SHADOW = '0px 1px 6px rgba(0, 0, 0, 0.9), 0px 8px 12px rgba(13, 14, 14, 0.8)'
+
+const ShareButtonDisplay = styled.div`
+  display: flex;
+  cursor: pointer;
+  position: relative;
+
+  &:hover {
+    color: ${({ theme }) => darken(0.1, theme.text2)};
+  }
+`
+const ShareActions = styled.div`
+  position: absolute;
+  top: 28px;
+  right: 0px;
+  padding: 8px 0px;
+  display: flex;
+  flex-direction: column;
+  width: fit-content;
+  overflow: auto;
+  background-color: ${({ theme }) => theme.bg0};
+  border: 1px solid ${({ theme }) => theme.bg1};
+  box-shadow: ${SHADOW};
+  border-radius: 12px;
+`
+const ShareAction = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 12px 16px;
+  gap: 8px;
+  width: 200px;
+  height: 48px;
+  color: ${({ theme }) => theme.text1};
+  cursor: pointer;
+
+  &:hover {
+    color: ${({ theme }) => darken(0.08, theme.text1)};
+    background-color: ${({ theme }) => darken(0.08, theme.bg1)};
+  }
+`
+const LinkCopied = styled.div<{ show: boolean }>`
```

Comment:
this probably should be a component that lives outside of this part of the codebase, can you add a todo

Reply:
I think this might, actually! We have some "click-to-copy" interactions already. I'd look around for those @kayleegeorge